discovery_engine/lib/src/domain/models/active_search.dart

```diff
@@ -12,28 +12,46 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:equatable/equatable.dart';
 import 'package:hive/hive.dart'
     show HiveType, HiveField, TypeAdapter, BinaryReader, BinaryWriter;
+import 'package:xayn_discovery_engine/src/api/models/active_search.dart' as api;
 import 'package:xayn_discovery_engine/src/domain/repository/type_id.dart'
     show searchTypeId, searchByTypeId;
 
-part 'active_search.freezed.dart';
 part 'active_search.g.dart';
 
 /// [ActiveSearch] represents attributes of a performed search.
-@freezed
-class ActiveSearch with _$ActiveSearch {
-  @HiveType(typeId: searchTypeId)
-  const factory ActiveSearch({
-    @HiveField(0) required String searchTerm,
-    @HiveField(1) required int requestedPageNb,
-    @HiveField(2) required int pageSize,
-    @HiveField(3, defaultValue: SearchBy.query) required SearchBy searchBy,
-  }) = _ActiveSearch;
-
-  factory ActiveSearch.fromJson(Map<String, Object?> json) =>
-      _$ActiveSearchFromJson(json);
+@HiveType(typeId: searchTypeId)
+class ActiveSearch with EquatableMixin {
+  @HiveField(0)
+  final String searchTerm;
+  @HiveField(1)
+  final int requestedPageNb;
+  @HiveField(2)
+  final int pageSize;
+  @HiveField(3, defaultValue: SearchBy.query)
+  final SearchBy searchBy;
+
+  const ActiveSearch({
+    required this.searchTerm,
+    required this.requestedPageNb,
+    required this.pageSize,
+    required this.searchBy,
+  });
+
+  ActiveSearch nextPageSearch() => ActiveSearch(
```

Comment:
:art: the naming seems a bit redundant
```suggestion
  ActiveSearch nextPage() => ActiveSearch(
```